op-node/rollup/derive/span_batch.go

```diff
@@ -505,7 +505,7 @@ func (b *SpanBatch) peek(n int) *SpanBatchElement { return b.Batches[len(b.Batch
 // updates l1OriginCheck or parentCheck if needed.
 func (b *SpanBatch) AppendSingularBatch(singularBatch *SingularBatch, seqNum uint64) error {
 	// if this new element is not ordered with respect to the last element, panic
-	if len(b.Batches) > 0 && b.peek(0).Timestamp > singularBatch.Timestamp {
+	if len(b.Batches) > 0 && b.peek(0).Timestamp >= singularBatch.Timestamp {
```

Comment:
I think this case would only happen if we added a batch twice which I think we may actually want to prevent cc @axelKingsley

Reply:
I agree, adding the same batch twice would be incorrect and is a valid panic condition. 

I still need to read over the rest of this PR, specifically the revert code. It's a good feature to add but it can be tricky to ensure the bitmaps get reverted correctly.